nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/SAMLAccessResource.java

```diff
@@ -0,0 +1,650 @@
+        cookie.setMaxAge(60);
+        cookie.setSecure(true);
+        httpServletResponse.addCookie(cookie);
+
+        // get the state for this request
+        final String relayState = samlStateManager.createState(samlRequestIdentifier);
+
+        // initiate the login request
+        try {
+            samlService.initiateLogin(httpServletRequest, httpServletResponse, relayState);
+        } catch (Exception e) {
+            forwardToLoginMessagePage(httpServletRequest, httpServletResponse, e.getMessage());
+            return;
+        }
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+    @Produces(MediaType.WILDCARD)
+    @Path(SAMLEndpoints.LOGIN_CONSUMER_RELATIVE)
+    @ApiOperation(
+            value = "Processes the SSO response from the SAML identity provider for HTTP-POST binding.",
+            notes = NON_GUARANTEED_ENDPOINT
+    )
+    public void samlLoginHttpPostConsumer(@Context HttpServletRequest httpServletRequest,
+                                          @Context HttpServletResponse httpServletResponse,
+                                          MultivaluedMap<String, String> formParams) throws Exception {
+
+        // only consider user specific access over https
+        if (!httpServletRequest.isSecure()) {
+            forwardToLoginMessagePage(httpServletRequest, httpServletResponse, AUTHENTICATION_NOT_ENABLED_MSG);
+            return;
+        }
+
+        // ensure saml is enabled
+        if (!samlService.isSamlEnabled()) {
+            forwardToLoginMessagePage(httpServletRequest, httpServletResponse, SAMLService.SAML_SUPPORT_IS_NOT_CONFIGURED);
+            return;
+        }
+
+        // process the response from the idp...
+        final Map<String, String> parameters = getParameterMap(formParams);
+        samlLoginConsumer(httpServletRequest, httpServletResponse, parameters);
+    }
+
+    @GET
+    @Consumes(MediaType.WILDCARD)
+    @Produces(MediaType.WILDCARD)
+    @Path(SAMLEndpoints.LOGIN_CONSUMER_RELATIVE)
+    @ApiOperation(
+            value = "Processes the SSO response from the SAML identity provider for HTTP-REDIRECT binding.",
+            notes = NON_GUARANTEED_ENDPOINT
+    )
+    public void samlLoginHttpRedirectConsumer(@Context HttpServletRequest httpServletRequest,
+                                              @Context HttpServletResponse httpServletResponse,
+                                              @Context UriInfo uriInfo) throws Exception {
+
+        // only consider user specific access over https
+        if (!httpServletRequest.isSecure()) {
+            forwardToLoginMessagePage(httpServletRequest, httpServletResponse, AUTHENTICATION_NOT_ENABLED_MSG);
+            return;
+        }
+
+        // ensure saml is enabled
+        if (!samlService.isSamlEnabled()) {
+            forwardToLoginMessagePage(httpServletRequest, httpServletResponse, SAMLService.SAML_SUPPORT_IS_NOT_CONFIGURED);
+            return;
+        }
```

Comment:
This logic is repeated in several other methods, what do you think about refactoring it to a shared method that returns a boolean to determine whether to proceed?  Perhaps something along the lines of `boolean isSamlEnabled(HttpServletRequest, HttpServletResponse)`?